app/components/Welcome.js

```diff
@@ -15,32 +26,63 @@ export default class Welcome extends Component {
     this.props.stopUpdateLang()
   }
 
-  goToNext () {
+  // Called when user clicks on next button.
+  onSaveSettings () {
+    // Saving new Language.
+    let newSettings = {
+      language: this.state.language
+    }
+    for (let setting in this.state) {
+      // Avoid adding same settings as a new settings.
+      if (this.state[setting] !== this._defaults[setting]) {
+        newSettings[setting] = this.state[setting]
+      }
+    }
+
+    this.props.saveSettings(newSettings)
     setNotFirstTimeFlag()
   }
 
+  onLanguageChange (event) {
+    this.state.language = event.target.value
```

Comment:
I tried using the ``` this.setState({language: event.target.value}) ``` function but the setState get bugged and doesn't work correctly, only when I assign the state like that works